<launch>

  <arg name="input/image" default="image_raw" />
  <arg name="input/rois" default="rois" />
  <arg name="output/traffic_light_states" default="traffic_light_states" />

  <arg name="classifier_type" default="cnn" />
  <!-- <arg name="model_file_path" default="$(find traffic_light_classifier)/data/mobilenetv2.onnx" /> -->
  <arg name="model_file_path" default="$(find traffic_light_classifier)/data/efficientnet.onnx" />
  <arg name="label_file_path" default="$(find traffic_light_classifier)/data/lamp_labels.txt" />
  <arg name="precision" default="fp16" />

  <node name="traffic_light_classifier" pkg="traffic_light_classifier" type="traffic_light_classifier_node" output="screen">
    <remap from="~input/image" to="$(arg input/image)" />
    <remap from="~input/rois" to="$(arg input/rois)" />
    <remap from="~output/traffic_light_states" to="$(arg output/traffic_light_states)" />
    <param name="approximate_sync" value="false" />
    <param name="classifier_type" value="$(arg classifier_type)" />
    <param name="model_file_path" value="$(arg model_file_path)" />
    <param name="label_file_path" value="$(arg label_file_path)" />
    <param name="precision" value="$(arg precision)" />
  </node>

</launch>
